### PR TITLE
IMPRO-1328: Improved documentation for IMPROVER

### PR DESCRIPTION
**Pull request by bayliffe**
Adds some general information about IMPROVER, along with some slightly improved help for those wishing to use the code. Includes links to public presentations about IMPROVER and a few example output images.

make-clean also modified to clear autogenerated rst files (source/improver.*.rst) such that after a make clean the documentation is entirely remade from the code doc-strings.

The resulting documentation can be seen here: https://bens-improver-fork.readthedocs.io/en/doctesting/index.html

And can be compared with the current documentation here: https://improver.readthedocs.io/en/latest/

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: doc/source/about.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
+What is IMPROVER?
+-----------------
+
+IMPROVER: Integrated Model post-PROcessing and VERification
+
+The Met Office in the UK is leading an open source project to create a probabilistic post processing system for use with ensemble weather prediction models. The system is designed to extract the wealth of information provided by ensemble forecasts and present it in a format that is suitable for the production of automated weather forecasts, and for use by forecasters and the public.
```
Comment (PaulAbernethy) [inline doc/source/about.rst]: I'd hyphenate post processing.

File: doc/source/about.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+   Probability of exceeding wind speed thresholds
+
+
+Structure of IMPROVER
+---------------------
+
+IMPROVER is designed as a modular post-processing system. The final product is created through the application of a sequence of processing steps, where the sequence can be readily modified to achieve different outcomes. The output of each step in the chain can be written out, allowing for verification against observations at each stage (using a suitable verification package, which is not part of IMPROVER). This enables the user to determine whether a given step in the chain is improving or harming the forecast quality.
+
+Any given step in the processing can be applied using the included command line interfaces (CLIs). A complex system is built be calling the CLIs in sequence. The simple schematic below gives an example processing chain.
```
Comment (PaulAbernethy) [inline doc/source/about.rst]: Typo: 'built be calling' => 'built by calling'

File: doc/source/frontpage.rst [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+   :alt: License
+.. image:: https://img.shields.io/badge/python-3.6-blue.svg
+   :target: https://www.python.org/downloads/release/python-360/
+   :alt: Python Version
+.. image:: https://travis-ci.org/metoppv/improver.svg?branch=master
+   :target: https://travis-ci.org/metoppv/improver
+   :alt: Build Status
+
+
+IMPROVER_ is an open source project being led by the Met Office in the UK to create a probabilistic post processing system for ensemble weather prediction models. The aim is to extract the best possible probabilistic forecasts from multiple ensemble models, combining data from convection permitting and parameterised forecasts into a seamless blend.
```
Comment (PaulAbernethy) [inline doc/source/frontpage.rst]: post-processing

File: doc/source/about.rst (hunk 1 above)
Comment (bayliffe) [reply doc/source/about.rst]: Done.

File: doc/source/about.rst (hunk 2 above)
Comment (bayliffe) [reply doc/source/about.rst]: Done.

File: doc/source/frontpage.rst (hunk 3 above)
Comment (bayliffe) [reply doc/source/frontpage.rst]: Done.

File: doc/source/about.rst [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+Within the Met Office IMPROVER is being used to provide a new probabilistic approach to post-processing forecasts. Gridded forecasts taken from multiple configurations of the Met Office's Unified Model (UM) first undergo physical corrections if necessary, before being converted into probabilities. The resulting probabilistic forecasts from each UM configuration are then blended to create seamless probabilistic outputs. These gridded probabilities aim to deliver both improved forecast information at each grid square and at geographical (“spot”) locations.
```
Comment (MoseleyS) [inline doc/source/about.rst]:
```suggestion
Within the Met Office, IMPROVER is being used to provide a new probabilistic approach to post-processing forecasts. Gridded forecasts taken from multiple configurations of the Met Office's Unified Model (UM) first undergo physical corrections if necessary, before being converted into probabilities. The resulting probabilistic forecasts from each UM configuration are then blended to create seamless probabilistic outputs. These gridded probabilities aim to deliver improved forecast information both at each grid square and at geographical (“spot”) locations.
```

Comment (MoseleyS) [reply doc/source/about.rst]: Add a link to the UM (https://www.metoffice.gov.uk/research/approach/modelling-systems/unified-model/index)

File: doc/source/about.rst [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+   Probability of exceeding wind speed thresholds
+
+
+Structure of IMPROVER
+---------------------
+
+IMPROVER is designed as a modular post-processing system. The final product is created through the application of a sequence of processing steps, where the sequence can be readily modified to achieve different outcomes. The output of each step in the chain can be written out, allowing for verification against observations at each stage (using a suitable verification package, which is not part of IMPROVER). This enables the user to determine whether a given step in the chain is improving or harming the forecast quality.
+
+Any given step in the processing can be applied using the included command line interfaces (CLIs). A complex system is built by calling the CLIs in sequence. The simple schematic below gives an example processing chain.
```
Comment (MoseleyS) [inline doc/source/about.rst]: Add link from CLIs to "Using IMPROVER" paragraph.

File: doc/source/about.rst [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+.. code:: console
+
+    bin/improver threshold input_file.nc output_file.nc 5 10 15 20 --threshold_units m/s
+
+
+* input_file.nc is a netCDF file containing a forecast diagnostic data cube, e.g. wind speeds across an x-y grid at a given time.
+* for each threshold specified (5, 10, 15, 20 m/s) a new x-y grid of data will be created. Each point in the grid will contain a 0 if the input wind speed at that point was below the threshold, or 1 if it was above the threshold.
+* output_file.nc will be a new netCDF file containing the resulting data cube with an additional leading dimension that corresponds to the given thresholds (5, 10, 15, 20 m/s).
+
+This simple example covers one step in a processing chain. Additional information about using any CLI can be found on the command line using `-h`, e.g.:
```
Comment (MoseleyS) [inline doc/source/about.rst]: Use the more human-readable `--help`, it seems more appropriate for documentation.

File: doc/source/about.rst [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+.. code:: console
+
+    bin/improver nbhood -h
+
```
Comment (MoseleyS) [inline doc/source/about.rst]: Add: "The current list of CLIs can be found \<a href='improver.cli.html'>here</a>"

File: doc/source/about.rst (hunk 4 above)
Comment (bayliffe) [reply doc/source/about.rst]: Done and done.

File: doc/source/about.rst (hunk 5 above)
Comment (bayliffe) [reply doc/source/about.rst]: Done.

File: doc/source/about.rst (hunk 6 above)
Comment (bayliffe) [reply doc/source/about.rst]: Done.

File: doc/source/about.rst (hunk 7 above)
Comment (bayliffe) [reply doc/source/about.rst]: Done.

**Comment by bayliffe** [comment]
@MoseleyS  I linked the Met Office text, I have not added a logo as I don't want to get embroiled in the proper way to use Met Office branding.

**Pull request merged**